workspaces/arborist/lib/shrinkwrap.js

```diff
@@ -979,8 +1003,10 @@ class Shrinkwrap {
     // if we don't have either, just an empty object so nothing matches below.
     // This will effectively just save the version and resolved, as if it's
     // a standard version/range dep, which is a reasonable default.
-    const spec = !edge ? rSpec
-      : npa.resolve(node.name, edge.spec, edge.from.realpath)
+    let spec = rSpec
+    if (edge) {
+      spec = npa.resolve(node.name, edge.spec, edge.from.realpath)
+    }
```

Comment:
```suggestion
    const spec = edge ? npa.resolve(node.name, edge.spec, edge.from.realpath) : rSpec
```

I 💯% appreciate the de-ternaryifying happenening in this PR. But that does make me want to call out that I think in this case the ternary is easier for me to read. I'm not asking for changes here, just showing my preference for the ternary operator in some cases.

Reply:
Would an if/else be better here? Is it the overloading of the `let` declaration?

Reply:
It's mostly the order of operations. My inline suggestion is even different than the original ternary because it reverses the order. This is getting in the weeds, but I read the ternary as: `if we have an edge, resolve from edge, otherwise its rSpec` where the if statement I read as `spec is usually rSpec, but if we have an edge, then resolve from the edge`.

I think even using `let spec` and adding an else branch reads better to me. But when writing code like that I usually opt for the ternary because then I don't have to name another intermediate variable. I think naming fatigue is real for me, so it's not that the let declaration is overloaded, it's that I have to Name One More Thing.

I think it should be kept as you have it in the PR since it matches the rest of the file which was in need of this refactor. This was just an excuse to give a little love that I have for ternaries.

Reply:
Rules are guidelines and should serve their purpose. If we find a case where ternaries are easier to read then let's do it.  Use them in *your* prs and let's find out.